src/backend/opencl/err_opencl.hpp

```diff
@@ -22,10 +23,13 @@
 
 namespace opencl {
 template<typename T>
-void verifyDoubleSupport() {
+void verifyTypeSupport() {
     if ((std::is_same<T, double>::value || std::is_same<T, cdouble>::value) &&
         !isDoubleSupported(getActiveDeviceId())) {
         AF_ERROR("Double precision not supported", AF_ERR_NO_DBL);
+    } else if (std::is_same<T, common::half>::value &&
+               !isHalfSupported(getActiveDeviceId())) {
+        AF_ERROR("Half precision not supported", AF_ERR_NO_HALF);
     }
```

Comment:
My comment on #2415 , https://github.com/arrayfire/arrayfire/pull/2415#discussion_r249435163 , is more of concern  especially in OpenCL backend, don't you think so ? We can't assume if double support is there, there is half support.

Hm, is this okay for now because we only ammulate half computation using float for actual computation ?

Reply:
Nevermind, I missed the type match validation. My bad.